### ISSUE TITLE
Does Simplifyify have a programmatic API (for use in Gulp, Grunt, etc.)?

### ISSUE THREAD

**Issue by BigstickCarpet** [BigstickCarpet/simplifyify#3]
This question was asked by @nmccready on [another issue](https://github.com/substack/node-browserify/issues/1233#issuecomment-140244008).  I'm answering it here so it's easy for other people to find as well.

The answer is, "Yes!".  Simplifyify _does_ have a programmatic API.  You _aren't_ limited to using its command-line interface.   In fact, the CLI is simply a very thin wrapper around the programmatic API. [Here's the API](https://github.com/BigstickCarpet/simplifyify/blob/eba983197e0512619c8818b53f7752e795b42b0b/lib/index.js#L9-L28), and [here's sample code](https://github.com/BigstickCarpet/simplifyify/blob/ce6b47b94f524ed9e66396fb3b5b951b655328ad/bin/simplifyify.js#L51-L75).  Basically, you just pass an array of strings (file paths and/or glob patterns) and an options param.  You get back an [`EventEmitter`](https://nodejs.org/api/events.html#events_class_events_eventemitter), which fires all the Browserify & Watchify events.

```javascript
var simplifyify = require("simplifyify");

gulp.task("browserify", function(done) {
  simplifyify("lib/*.module.js", 
    { 
        outfile: "dist/*.bundle.js", 
        debug: true,
        minify: true 
    })
    .on("end", function() {
        // Finished successfully!
        done();
    })
    .on("error", function(err) {
        // Something went wrong
        done(err);
    });
});
```

**Comment by BigstickCarpet** [comment]
I'm leaving this issue open until I get around to documenting the API in the README

**Comment by nmccready** [comment]
ty

**Comment by BigstickCarpet** [comment]
:+1: sure thing.  Sorry it took so long to get around to it.  The joys of being the sole maintainer of several open-source projects :-/

**Comment by nmccready** [comment]
I sure know how that goes; not fun when it is large at scale

**Comment by nmccready** [comment]
I'm trying to remember how I got here; this is the workaround sourcemaps to get the raw js to debug unlike browserify right? I

**Comment by BigstickCarpet** [comment]
yeah.  it extracts the sourcemaps to external files, so they work better with most debugging tools

**Comment by nmccready** [comment]
Cool; yeah I need to do this as I was waiting on your reply; tired of debugging coffee in the browser.
